e2e_tests/tests/cluster/test_users.py

```diff
@@ -57,81 +72,62 @@ def det_run(args: List[str]) -> str:
     return cast(str, pexpect.run(f"det -m {conf.make_master_url()} {' '.join(args)}").decode())
 
 
-def log_in_user(credentials: authentication.Credentials) -> int:
+def log_in_user(credentials: authentication.Credentials, expectedStatus: int = 0) -> None:
+    import re
+
     username, password = credentials
     child = det_spawn(["user", "login", username])
     child.setecho(True)
-    expected = f"Password for user '{username}':"
-    child.expect(expected, timeout=EXPECT_TIMEOUT)
-    child.sendline(password)
-    child.read()
-    child.wait()
-    return cast(int, child.exitstatus)
-
-
-def create_user(n_username: str, admin_credentials: authentication.Credentials) -> None:
-    a_username, a_password = admin_credentials
-
-    child = det_spawn(["-u", a_username, "user", "create", n_username])
+    expected = re.escape(f"Password for user '{username}': ")
 
-    expected_password_prompt = f"Password for user '{a_username}':"
-    i = child.expect([expected_password_prompt, pexpect.EOF], timeout=EXPECT_TIMEOUT)
-    if i == 0:
-        child.sendline(a_password)
+    child.expect(expected, EXPECT_TIMEOUT)
+    child.sendline(password)
     child.read()
     child.wait()
     child.close()
+    assert child.exitstatus == expectedStatus
 
-    assert child.exitstatus == 0
+
+def create_test_user(add_password: bool = False) -> authentication.Credentials:
+    n_username = get_random_string()
+    command = ["det", "-m", conf.make_master_url(), "user", "create", n_username]
+    output = subprocess.check_call(command)
+    assert output == 0
```

Comment:
`run(..., check=True)` or `check_call` will raise if anything is wrong
```suggestion
    subprocess.run(command, check=True)
```